UserService/Repository/UserRepository.cs

```diff
@@ -12,41 +13,57 @@ public UserRepository(DataContext dbContext)
     {
         _dbContext = dbContext;
     }
-    public bool CreateUser(User user)
+
+    public void CreateUser(User user)
     {
         _dbContext.Add(user);
-        return Save();
+        Save();
     }
 
     public IEnumerable<User> GetAllUsers()
     {
-        return _dbContext.Users.AsEnumerable();
+        var users = _dbContext.Users.AsEnumerable();
+        return users;
     }
 
     public User GetUser(string id)
     {
-        return _dbContext.Users.Where(u => u.Id == id).FirstOrDefault();
+        var user = _dbContext.Users.Where(u => u.Id == id).FirstOrDefault();
+
+        if (user == null)
+        {
+            throw new EntityNotFoundException($"User with ID {id} not found.");
+        }
+
+        return user;
     }
 
-    public bool Save()
+    public void Save()
     {
-        return _dbContext.SaveChanges() > 0;
+        try
```

Comment:
Didn't we say we were going to use try-catch blocks in the controller instead of the repository? It's cleaner, isn't it?

Reply:
yes, but I noticed that if we add the try catch blocks in the controller, we'll have code duplication